invokeai/backend/install/model_install_backend.py

```diff
@@ -52,126 +46,349 @@
-    global Datasets
-    if Datasets:
-        return Datasets
-    return (Datasets := OmegaConf.load(Dataset_path)['diffusers'])
-
-def install_requested_models(
-        diffusers: ModelInstallList = None,
-        controlnet: ModelInstallList = None,
-        lora: ModelInstallList = None,
-        ti: ModelInstallList = None,
-        cn_model_map: Dict[str,str] = None, # temporary - move to model manager
-        scan_directory: Path = None,
-        external_models: List[str] = None,
-        scan_at_startup: bool = False,
-        precision: str = "float16",
-        purge_deleted: bool = False,
-        config_file_path: Path = None,
-        model_config_file_callback:  Callable[[Path],Path] = None
-):
-    """
-    Entry point for installing/deleting starter models, or installing external models.
-    """
-    access_token = HfFolder.get_token()
-    config_file_path = config_file_path or default_config_file()
-    if not config_file_path.exists():
-        open(config_file_path, "w")
-
-    # prevent circular import here
-    from ..model_management import ModelManager
-    model_manager = ModelManager(OmegaConf.load(config_file_path), precision=precision)
-    if controlnet:
-        model_manager.install_controlnet_models(controlnet.install_models, access_token=access_token)
-        model_manager.delete_controlnet_models(controlnet.remove_models)
-
-    if lora:
-        model_manager.install_lora_models(lora.install_models, access_token=access_token)
-        model_manager.delete_lora_models(lora.remove_models)
-
-    if ti:
-        model_manager.install_ti_models(ti.install_models, access_token=access_token)
-        model_manager.delete_ti_models(ti.remove_models)
-
-    if diffusers:
-        # TODO: Replace next three paragraphs with calls into new model manager
-        if diffusers.remove_models and len(diffusers.remove_models) > 0:
-            logger.info("Processing requested deletions")
-            for model in diffusers.remove_models:
-                logger.info(f"{model}...")
-                model_manager.del_model(model, delete_files=purge_deleted)
-            model_manager.commit(config_file_path)
-
-        if diffusers.install_models and len(diffusers.install_models) > 0:
-            logger.info("Installing requested models")
-            downloaded_paths = download_weight_datasets(
-                models=diffusers.install_models,
-                access_token=None,
-                precision=precision,
+        # first populate with the entries in INITIAL_MODELS.yaml
+        for key, value in self.datasets.items():
+            name,base,model_type = ModelManager.parse_key(key)
+            value['name'] = name
+            value['base_type'] = base
+            value['model_type'] = model_type
+            model_dict[key] = ModelLoadInfo(**value)
+
+        # supplement with entries in models.yaml
+        installed_models = self.mgr.list_models()
+        for base in installed_models.keys():
```

Comment:
```suggestion
        for base in installed_models:
```
Had to do this change to get it running on a clean install with invokeai-configure

Reply:
Thanks. This was due to a merge clash. Now fixed.